src/SearchBar.jsx

```diff
@@ -1,66 +1,168 @@
-import React from 'react';
+import PropTypes from 'prop-types'
+import React from 'react'
 import { connect } from 'react-redux'
 import search from './actions/search'
-import './SearchBar.less';
+import Typeahead, { MODE_OPEN } from './Typeahead'
+import HighlightingOption from './Typeahead/HighlightingOption'
+import './SearchBar.less'
+
+const searchFields = {
+  'all': 'All Data',
+  'company': 'Company Name',
+  'complaint_what_happened': 'Narratives'
+}
 
 export class SearchBar extends React.Component {
   constructor(props) {
-    super(props);
+    super(props)
     this.state = {
-      inputValue: props.searchText
-    };
+      inputValue: props.searchText,
+      searchField: props.searchField
+    }
 
     // This binding is necessary to make `this` work in the callback
     // https://facebook.github.io/react/docs/handling-events.html
-    this._handleSubmit = this._handleSubmit.bind(this);
-    this._updateInputValue = this._updateInputValue.bind(this);
+    this._handleSubmit = this._handleSubmit.bind(this)
```

Comment:
It might be nice to have a `bindAll` utility function. https://www.npmjs.com/package/lodash.bindall